docs/reference/api/storage/FlashSimBlockDevice.md

```diff
@@ -0,0 +1,28 @@
+## FlashSimBlockDevice
+
+<span class="images">![](https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_slicing_block_device.png)<span>FlashSimBlockDevice class hierarchy</span></span>
+
+The FlashSimBlockDevice class is a block device adpator, whose purpose is to simulate the behavior of a flash component if the underlying block device doesn't support such a behavior. This includes the following:
```

Comment:
Query: What are the use cases for this API? When would a user want to use this API rather than some of the other block devices?